azkaban-common/src/main/java/azkaban/imagemgmt/models/ImageRampRule.java

```diff
@@ -1,35 +1,62 @@
+/*
+ * Copyright 2022 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
 package azkaban.imagemgmt.models;
 
-import java.util.List;
 import java.util.Set;
+import javax.annotation.Nullable;
 
 
+/**
+ * This class represents Image Ramp Rule metadata.
+ */
 public class ImageRampRule extends BaseModel {
   // ramp rule name
-  String ruleId;
+  String ruleName;
   // image name of ramp rule for
   String imageName;
   // image version of ramp rule for
   String imageVersion;
   // rule owners
   Set<String> owners;
-  // Ramp Rule for HP flows or not
+  // Ramp Rule for HP flows
+  // ture if rule is for HP flow; false otherwise
   boolean isHPRule;
 
-  public ImageRampRule(String ruleId,
+  public ImageRampRule(String ruleName,
                 String imageName,
                 String imageVersion,
                 Set<String> owners,
-                boolean isHP) {
-    this.ruleId = ruleId;
+                boolean isHP,
+                String createdBy,
+                @Nullable String createdOn ,
```

Comment:
Why is this field Nullable?

Reply:
By mistake, initially I thought a different constructor then later I abandoned it and forgot to remove the annotation. It's fixed.